users_app/tests/test_permissions.py

```diff
@@ -33,63 +45,144 @@ def test_group_creation(self):
+        with patch('users_app.pipeline.pipeline.get_ap_allowed_accounts', return_value=ALLOWED_AP_ACCOUNTS_FOR_TEST):
+            add_user_to_group(True, user)
 
-        self.assertFalse(self.supplier_group in user.groups.all())
-        self.assertFalse(self.ap_reporter_group in user.groups.all())
+        self.assertTrue(in_group(self.ap_admin_group, user))
+        self.assertTrue(in_group(self.ap_manager_group, user))
 
-    def test_AP_already_in_ap_admin_and_ap_manager_group_are_not_re_assigned(self):
-        user = UserFactory(email='nahuel.valencia@eventbrite.com')
-        user.groups.add(self.ap_admin_group, self.ap_manager_group)
+        self.assertFalse(in_group(self.supplier_group, user))
+        self.assertFalse(in_group(self.ap_reporter_group, user))
+
+    def test_add_user_to_group_add_new_ap_to_reporter_group(self):
+        user = UserFactory(email='ap_not_in_the_list@eventbrite.com')
 
-        add_user_to_group(False, user)
+        with patch('users_app.pipeline.pipeline.get_ap_allowed_accounts', return_value=ALLOWED_AP_ACCOUNTS_FOR_TEST):
+            add_user_to_group(True, user)
 
-        self.assertTrue((self.ap_admin_group and self.ap_manager_group) in user.groups.all())
-        self.assertTrue((self.supplier_group and self.ap_reporter_group) not in user.groups.all())
+        self.assertTrue(in_group(self.ap_reporter_group, user))
 
-    def test_add_new_AP_to_ap_reporter_group(self):
-        user = UserFactory(email='ap_no_in_the_list@eventbrite.com')
-        add_user_to_group(True, user)
+        self.assertFalse(in_group(self.ap_admin_group, user))
+        self.assertFalse(in_group(self.ap_manager_group, user))
+        self.assertFalse(in_group(self.supplier_group, user))
 
-        self.assertTrue(self.ap_reporter_group in user.groups.all())
+    def test_add_user_to_group_add_new_supplier_to_a_supplier_group(self):
+        user = UserFactory(email='nahuel.valencia21@gmail.com')
 
-        self.assertFalse(self.ap_admin_group in user.groups.all())
-        self.assertFalse(self.ap_manager_group in user.groups.all())
-        self.assertFalse(self.supplier_group in user.groups.all())
+        with patch('users_app.pipeline.pipeline.get_ap_allowed_accounts', return_value=ALLOWED_AP_ACCOUNTS_FOR_TEST):
+            add_user_to_group(True, user)
 
-    def test_AP_already_in_ap_reporter_group_are_not_re_assigned(self):
-        user = UserFactory(email='ap_no_in_the_list@eventbrite.com')
-        user.groups.add(self.ap_reporter_group)
+        self.assertTrue(in_group(self.supplier_group, user))
 
-        add_user_to_group(False, user)
+        self.assertFalse(in_group(self.ap_admin_group, user))
+        self.assertFalse(in_group(self.ap_manager_group, user))
+        self.assertFalse(in_group(self.ap_reporter_group, user))
 
-        self.assertEqual(self.ap_reporter_group, user.groups.get())
+    def test_add_user_to_group_doesnt_add_ap_if_it_is_already_in_ap_admin_and_ap_manager_group(self):
```

Comment:
nit: Very long title, remember that you shouldn't pass the 180 characters. It doesn't matter now, it's a comment for the future